### PR TITLE
Add more unit tests

### PR DESCRIPTION
**Pull request by cschembor3**
# Overview

This PR is to add some more unit tests for the http module. This captures #13 .

## Tests added

In this PR tests for the following functions have been added:
- `req_from_uri`
- `set_query`
- `get_req_header`
- `prepend_req_header`
- `set_req_body`
- `set_method`
- `map_resp_body`
- `map_req_body`
- `try_map_resp_body`
- `default_req`
- `set_host`
- `set_path`

### CODE REVIEW

**Comment by lpil** [comment]
Looking good so far! Thank you for this contribution, really helps get us to that production quality :) 

Let me know when you would like a full review.

**Comment by cschembor-ww** [comment]
Sounds great! Just getting started with gleam, but trying to help out wherever I can!

**Comment by cschembor3** [comment]
Sounds great! Just getting started with gleam, but I'm excited to help out and contribute wherever I can!

File: test/gleam/http_test.gleam [hunk 1]
```diff
@@ -722,6 +745,195 @@ pub fn get_query_test() {
...
+    Ok("awesome"),
+    request
+    |> http.get_req_header(header_key))
+
+  let request = make_request([tuple("answer", "42")])
+  should.equal(
+    Error(Nil),
+    request
+    |> http.get_req_header(header_key))
+}
```
Comment (lpil) [inline test/gleam/http_test.gleam:797]: Could we write these assertions like this:

```
request
|> http.get_req_header(header_key)
|> should.equal(Error(Nil))
```

I think it is easier to read, and the "expected" value is intended to be the second argument.

**Review by lpil — COMMENTED** [review]
This looks great! Thank you!

I've left one note inline. Would it also be possible to run `gleam format` to unsure the code uses the standard format?

The commit history looks good to me :)

File: test/gleam/http_test.gleam (hunk 1 above)
Comment (cschembor3) [reply test/gleam/http_test.gleam:802]: Definitely, that makes sense. Updated to reflect this change.

**Comment by cschembor3** [comment]
I've fixed the formatting, and addressed the code review comment.

**Review by lpil — APPROVED** [review]
Fantastic! Thank you

**Pull request merged**